.editorconfig

```diff
@@ -9,7 +9,7 @@ charset = utf-8
 end_of_line = lf
 indent_style = space
 insert_final_newline = true
-#trim_trailing_whitespace = true  # as long as we don't enforce this, this option causes more trouble than it helps
+trim_trailing_whitespace = true
```

Comment:
This is a no-go. I am the one who added the `.editorconfig` originally in 4d2a248bd2b810cfe92e9e98b9360a88d84cada2 and disabled `trim_trailing_whitespace` in eb442df2dcc970b80395dc4a8434f3c99dcdf802 -- because reality is that this kind of thing is too disruptive as long as we don't combine it with a big "remove all trailing whitespace in all files" commit -- which in itself has the potential of being very disruptive, causing open PRs to have annoying merge conflicts.

Without such a cleanup PR, people with an editor honoring this setting will keep creating PRs with lots of unrelated whitespace changes in them. You dismiss this by saying

> This does not influence code diffs since we can always set git and github to ignore whitespace changes.

but I don't think it is that simple at all. First off, not everyone even knows about that feature, and it is not on by default. Secondly, affecting code review is only one (minor) effect of these whitespace changes. Far more serious is that once again it increases the risk of creating pointless merge conflicts for others.

Therefore I think the only realistic change at this point is this one:
```suggestion
trim_trailing_whitespace = false
```


After the "Begehung", if we find a time when there are few open PRs, I'd open to revisit this and considering doing a mass "remove trailing spaces PR", if the people doing that agree to afterwards go through all open PRs and resolve their merge conflicts (well, at least those which weren't there before already)